txpipe/data_types/base.py

```diff
@@ -284,9 +284,57 @@ class TextFile(DataFile):
 class YamlFile(DataFile):
     """
     A data file in yaml format.
+    The top-level object in TXPipe YAML
+    files should always be a dictionary.
     """
     suffix = 'yml'
 
+    def __init__(self, path, mode, extra_provenance=None, validate=True, load_mode='full'):
```

Comment:
`mode` is not an optional argument, but I don't see it given any of the places you changed, am I missing something?

Reply:
when you call `open_input` or `open_output` it automatically selects either `r` or `w` for the mode accordingly.  It can be a bit annoying, though, maybe we should switch to the default being `r`?

Reply:
no I think it is fine then, that just didn't show up in the files changed, so I was unclear. I think that is all good then.